contrib/win32/win32compat/w32fd.c

```diff
@@ -387,16 +387,24 @@ w32_pipe(int *pfds)
 }
 
 int
-w32_open(const char *pathname, int flags, ...)
+w32_open(const char *pathname, int flags, ... /* arg */)
 {
 	int min_index = fd_table_get_min_index();
 	struct w32_io* pio;
+	va_list valist;
+	u_short mode = 0;
 
 	errno = 0;
 	if (min_index == -1)
 		return -1;
+	if (flags & O_CREAT) {
```

Comment:
why is this code specific to 0_CREAT? this can be called from anywhere with a read flag and a specific mode parameters..